### PR TITLE
Fix desktop notifications

### PR DESCRIPTION
**Pull request by gausie**
I had forgotten that KoLmafia used to have desktop notifications in some events. I'm pretty sure that 5307bc6729c6f033e27adc1f116c9bffae8b92bd has never worked? There are a few bits of wrong logic that would sum up to never seeing a notification ever. Those are now fixed. Also, since all OSes can use this option, I added it to the options manager and added explanation that it enables desktop notifications too.

I've also added to the RuntimeLibrary to allow scripters to send notifications. I personally very much want to use this and this was my initial reason for looking into the notification system in the first place. It is already possible to disable desktop notifications blanketly, as well as banning KoLmafia.jar from sending notifications from your desktop's notification manager (at least on OSX). So I'm happy allowing this - the onus is already on script users to make sure scripts they run don't do anything malicious, and this is far less ruinous than the kind of access we trust scripts with every day.

### CODE REVIEW

File: src/net/sourceforge/kolmafia/swingui/OptionsFrame.java [hunk 1]
```diff
@@ -1478,7 +1478,10 @@ protected class UserInterfacePanel extends OptionsPanel implements Listener {
         System.getProperty("os.name").startsWith("Windows")
             ? new String[][] {
               {"guiUsesOneWindow", "Restrict interface to a single window"},
-              {"useSystemTrayIcon", "Minimize main interface to system tray"},
+              {
+                "useSystemTrayIcon",
+                "Minimize main interface to system tray (and enable desktop notifications)"
```
[the remark this answers is not in the document]
Comment (gausie) [reply src/net/sourceforge/kolmafia/swingui/OptionsFrame.java:1483]: No your inference is correct.

**Comment by gausie** [comment]
> A change or request for clarification.
> 
> > this is far less ruinous than the kind of access we trust scripts with every day.
> 
> I don't think this is true. Scripts can read and write files inside the Mafia directory and run arbitrary web commands. I guess you could use this to make a cryptominer, but you can't infect somebody with ransomware or steal their passwords / wallet key / arbitrary files outside the Mafia directory.

I guess I mean you can take all the stuff from a person's account. Which to me seems worse than annoying notifications.

> Notifications can be used to scam people into downloading dangerous programs. My mother recently received a notification (from a website) with the Windows logo, and a comment like "Windows has detected dangerous viruses! Go to for assistance". With notifications access, scripts can encourage users to take dangerous actions more easily than they already can.

You're not able to set images and all the notifications say KoLmavia rWHATEVER. I do think this is fine - check it out and see how it looks on your machine

**Review by midgleyc — DISMISSED** [review]
I can see notifications, and minimising to the tray works.

The tooltip is subtly incorrect:
> 6 days until Laboacute;r Day, Moxie bonus today and tomorrow.

the XML characters don't work ;)

For the preference, I would remove the brackets around "and enable desktop notifications" -- I expect that's the main reason to enable the preference?

**Comment by midgleyc** [comment]
Slightly too late, and unrelated to the notifications: on Windows, if you have "minimize to system tray" enabled, and log out, it never exits. You have to end the task.

It's possible that's just for me, or this feature isn't really used much.

**Pull request merged**